### PR TITLE
Sync ESA-PSA label files to a local directory

### PR DESCRIPTION
**Pull request by nutjob4life**
## 🗒️ Summary

Merge this to satisfy the sync part of [registry-legacy-solr#135](https://github.com/NASA-PDS/registry-legacy-solr/issues/135).

## ⚙️ Test Data and/or Report

```console
$ cat harvest.cfg
cat: harvest.cfg: No such file or directory
$ du -sh download
du: download: No such file or directory
$ .venv/bin/python3 bin/portal/pds-sync-api.py
INFO Writing harvest XML config to harvest.cfg
INFO Downloading labels from https://pds.mcp.nasa.gov/api/search/1/products to download
INFO Generating ESA-PSA products from https://pds.mcp.nasa.gov/api/search/1/products
```
(25 minutes later)
```console
$ cat harvest.cfg
<?xml version='1.0' encoding='UTF-8'?>
<harvest xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://github.com/NASA-PDS/harvest/blob/main/src/main/resources/conf/configuration.xsd">
  <registry auth="/path/to/auth/file">app://localhost.xml</registry>
  <load>
    <directories>
      <path>/Users/kelly/Documents/Clients/JPL/PDS/Development/nasa-pds/operations/download</path>
    </directories>
  </load>
  <fileInfo processDataFiles="true" storeLabels="true">
    <fileRef replacePrefix="/Users/kelly/Documents/Clients/JPL/PDS/Development/nasa-pds/operations/download" with="https://url/to/archive"/>
  </fileInfo>
  <autoGenFields/>
</harvest>
$ du -sh download
6.0M	download
$ find download -type f | wc -l
    1449
```

## ♻️ Related Issues

- [registry-legacy-solr#135](https://github.com/NASA-PDS/registry-legacy-solr/issues/135)

### CODE REVIEW

**Review by alexdunnjpl — APPROVED** [review]
Trust

File: bin/portal/pds-sync-api.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+PDS_XSD_URL = 'https://github.com/NASA-PDS/harvest/blob/main/src/main/resources/conf/configuration.xsd'
+PROD_SEARCH_API_URL = 'https://pds.mcp.nasa.gov/api/search/1/products'
+XML_SCHEMA_INSTANCE_URI = 'http://www.w3.org/2001/XMLSchema-instance'
+NS_MAP = {'xsi': XML_SCHEMA_INSTANCE_URI}
+
+_logger = logging.getLogger(__name__)
+_search_key = 'ops:Harvest_Info.ops:harvest_date_time'
+_query_page_size = 50
+_psa_query = '(product_class eq "Product_Context" and ops:Harvest_Info.ops:node_name like "PSA")'
```
Comment (jordanpadams) [inline bin/portal/pds-sync-api.py:28]: @nutjob4life going to update this query to include `Product_Collection` and `Product_Bundle`

**Review by jordanpadams — CHANGES_REQUESTED** [review]

**Review by jordanpadams — COMMENTED** [review]
@nutjob4life also, in terms of speed, do we thing narrowing down the fields returned would speed this up at all? Or is most of the overhead in the label downloads?

**Review by nutjob4life — COMMENTED** [review]
Looks fine to me

**Comment by jordanpadams** [comment]
Merging. Will create separate task to actually run this.

**Pull request merged**